pkg/federation/sdlmerge/interface_type_extending.go

```diff
@@ -6,13 +6,16 @@ import (
 	"github.com/jensneuse/graphql-go-tools/pkg/operationreport"
 )
 
-func newExtendInterfaceTypeDefinition() *extendInterfaceTypeDefinitionVisitor {
-	return &extendInterfaceTypeDefinitionVisitor{}
+func newExtendInterfaceTypeDefinition(n *normalizer) *extendInterfaceTypeDefinitionVisitor {
+	return &extendInterfaceTypeDefinitionVisitor{
+		normalizer: n,
+	}
 }
 
 type extendInterfaceTypeDefinitionVisitor struct {
 	*astvisitor.Walker
-	document *ast.Document
+	document   *ast.Document
+	normalizer *normalizer
```

Comment:
still do not see a reason to include a full normalizer in this visitor when you need only entitySet

`e.normalizer.isTypeEntity` operates only with entitySet so basically you could have `isTypeEntity` attached to entitySet smth like this

```
type EntitySet map[string]struct{}

func (e *EntitySet) isTypeEntity(...)
```